src/inputs/textArea.jsx

```diff
@@ -57,114 +100,191 @@ class TextArea extends React.Component {
+            onClick(event.target.value);
         }
     }
 
     onFocus(event) {
-        if (!_.isUndefined(this.props.onFocus)) {
-            this.props.onFocus(event);
+        const {
+            onFocus,
+        } = this.props;
+        const {
+            isFocused,
+        } = this.state;
+
+        if (isFunction(onFocus)) {
+            onFocus(event);
         }
 
-        this.setState({ isFocused: !this.state.isFocused });
+        this.setState({ isFocused: !isFocused });
     }
 
     onKeyDown(event) {
-        if (!_.isUndefined(this.props.onKeyDown)) {
-            this.props.onKeyDown(event);
+        const {
+            onKeyDown,
+        } = this.props;
+
+        if (isFunction(onKeyDown)) {
+            onKeyDown(event);
         }
     }
 
     render() {
-        const { autoHeight, className, columns,
-            disabled, error, fluid,
-            id, inverse, label, labelStyle,
-            maxHeight, maxLength, minHeight, minLength, name,
-            placeholder, required, resize, rows, style } = this.props;
+        const {
+            autoHeight,
+            className,
+            columns,
+            disable,
+            disabled,
+            error,
+            fluid,
+            id,
+            inverse,
+            label,
+            labelStyle,
+            maxHeight,
+            maxLength,
+            minHeight,
+            minLength,
+            name,
+            placeholder,
+            required,
+            resize,
+            rows,
+            style,
+            value,
+        } = this.props;
+        const {
+            isFocused,
+        } = this.state;
+        const isDisabled = disable || disabled;
         const containerClasses = ClassNames('ui', 'text-area', className, {
             'text-area-auto-height': autoHeight,
-            'text-area-disabled': disabled,
+            'text-area-disabled': isDisabled,
             'text-area-error': error,
             'text-area-fluid': fluid,
-            'text-area-has-value': this.state.value,
-            'text-area-focused': this.state.isFocused,
+            'text-area-has-value': value,
+            'text-area-focused': isFocused,
             'text-area-inverse': inverse,
         });
 
         return (
             <div className={containerClasses} style={style}>
-                {label ? (
+                {label && (
                     <label className="label" htmlFor={id} style={labelStyle}>
                         {label}
 
-                        {required && !this.state.value ? (
+                        {required && !value ? (
                             <span className="text-area-required-indicator">*</span>
                         ) : null}
                     </label>
-                ) : null}
+                )}
 
                 <div className="text-area-container">
                     <textarea
-                        disabled={disabled}
+                        disabled={isDisabled}
```

Comment:
This is _precisely_ why I had a preference for `disabled` (past tense) in our component API as well.   Better harmony with the underlying native stuffs.  But c'est la vie! ;-)

Reply:
It's kinda bitter sweet.

Reply:
Yeah, all good.  Consistency on our API is the main thing.
Resolving this thread to reduce PR noise.